# MIT License

# Copyright (c) 2020 Hongrui Zheng

# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:

# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.

# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.

gym_simulator:
  ros__parameters:
    # topics and namespaces
    ego_namespace: ''
    ego_scan_topic: scan
    ego_wheel_odom_topic: wheel_odom
    ego_drive_topic: drive
    
    # laserscan parameters
    scan_fov: 4.7
    scan_beams: 1079

    # map parameters
    map_path: '/sim_ws/src/99p_gym_sim/maps/tape_map_no_lanes'
    map_img_ext: '.png'

    # ego starting pose on map
    sx: 0.85
    sy: 0.85
    stheta: 0.0
    # teleop
    kb_teleop: False
    num_agent: 1

task_commander:
  ros__parameters:
    waypoints_file: '/sim_ws/install/task_commander/share/task_commander/data/waypoints.csv'
    initial_pose_x:  0.85
    initial_pose_y: 0.85
    initial_pose_orientation_z: 0.0
    initial_pose_orientation_w: 1.0